girder/cumulus/server/models/volume.py

```diff
@@ -109,6 +111,16 @@ def append_to_log(self, user, id, record):
         self.update({'_id': ObjectId(id)}, {'$push': {'log': record}})
         send_log_notification('volume', volume, record)
 
+    def update_volume(self, user, volume):
```

Comment:
I'm not sure this should be call update_volume, its not really updating the existing volume, its more overriding it. I would be tempted to call it save_volume. The updating code is really in the patch method.